server/controllers/games.ts

```diff
@@ -14,52 +39,61 @@ async function createGame(req, res) {
       price_per_head: data.price_per_head,
       contact_details: data.contact_details
     });
+
     res.status(201).json(game)
   } catch (error) {
     console.error('Error creating game:', error);
     res.status(500).json({ message: `Internal server issue: ${error}` })
   }
 }
 
-async function getGames(req, res) {
+async function getGames(req: Request, res: Response): Promise<void> {
   try {
-    const games = await Game.find().populate('venue').populate('players')
-    res.status(200).json(games)
+    const games = await Game.find().populate('venue').populate('players');
+    res.status(200).json(games);
   } catch (error) {
     console.error('Error fetching games:', error);
-    res.status(500).json({ message: `Internal server issue: ${error}` })
+    res.status(500).json({ message: `Internal server issue: ${error}` });
   }
 }
 
-async function getGame(req, res) {
+
+
+async function getGame(req: Request, res: Response): Promise<void> {
   try {
-    const gameId = req.params.gameId
-    const game = await Game.findById(gameId).populate('venue').populate('players')
-    res.status(200).json(game)
+    const { gameId } = req.params;
+    const game = await Game.findById(gameId).populate('venue').populate('players');
+    if (!game) {
+      res.status(404).json({ message: "Game not found" });
+      return;
+    }
+    res.status(200).json(game);
   } catch (error) {
     console.error('Error fetching game:', error);
-    res.status(500).json({ message: `Internal server issue: ${error}` })
+    res.status(500).json({ message: `Internal server issue: ${error}` });
   }
 }
 
-async function joinGame(req, res) {
-  const gameId = req.params.gameId;
-  const playerUsername = req.body.userName;
+async function joinGame(req: JoinGameRequest, res: Response): Promise<void> {
+  const { gameId } = req.params;
+  const { userName: playerUsername } = req.body;
```

Comment:
`const { userName } = req.body`

Reply:
@hadeelbk there's two errors in this file now. Did you mean to do `const { userName: playerUsername } = req.body;` instead? Otherwise we have to update the property name in the type.
Also, on line 82 it expects userName, we have to change that to playerUsername now.
@mleker what was the purpose of this renaming? Just for clarity?

Reply:
otherwise it is just a redundancy with no reason
```suggestion

async function joinGame(req: JoinGameRequest, res: Response): Promise<void> {
  const { gameId } = req.params;
  const { userName } = req.body;

  try {
    const playerDetail = await Users.findOne({ userName });
```